docker/indexer/storage/storage.go

```diff
@@ -125,7 +107,8 @@ func (s *Store) Exists(ctx context.Context, addr string, hashType string, hash p
 	if _, ok := s.cache.Load(fmt.Sprintf(docKeyFmt, addr, hashType, hash)); ok {
 		return true, nil
 	}
-	key := datastore.NameKey(docKind, fmt.Sprintf(docKeyFmt, addr, hashType, hash), nil)
+	// hash[:], the [:] is important, since the formatting uses %x, which will return a different result if not used
```

Comment:
Can you explain this a bit more? Does this change the type which is why it's needed?

Reply:
Added more comments here to explain it, basically plumbing.Hash already implements it's own String() method, so this is formating it into hex twice.